From 4839da2e8157c4edf72f5adbedb5121f2728dbe7 Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@frugalware.org>
Date: Sun, 24 Mar 2013 09:50:47 +0100
Subject: [PATCH] service: no need to drop rc. prefix anymore

This reverts commit f5c88ec1330b61787441156de7d764a140774bd2. It is no
longer necessary, and adds unnecessary magic.
---

diff -Naur systemd-197.orig/src/core/service.c systemd-197/src/core/service.c
--- systemd-197.orig/src/core/service.c	2013-01-07 22:50:49.068315533 +0100
+++ systemd-197/src/core/service.c	2013-03-24 09:58:58.159324478 +0100
@@ -328,9 +328,6 @@
         if (endswith(name, ".sh"))
                 /* Drop Debian-style .sh suffix */
                 strcpy(stpcpy(r, name) - 3, ".service");
-        if (startswith(name, "rc."))
-                /* Drop Frugalware-style rc. prefix */
-                strcpy(stpcpy(r, name + 3), ".service");
         else
                 /* Normal init scripts */
                 strcpy(stpcpy(r, name), ".service");
@@ -992,10 +989,9 @@
         assert(s);
         assert(name);
 
-        /* For SysV services we strip the rc.* and *.sh
-         * prefixes/suffixes. */
-        if (startswith(name, "rc.") ||
-            endswith(name, ".sh.service"))
+        /* For SysV services we strip the *.sh
+         * suffixes. */
+        if (endswith(name, ".sh.service"))
                 return -ENOENT;
 
         STRV_FOREACH(p, UNIT(s)->manager->lookup_paths.sysvinit_path) {
@@ -1018,19 +1014,6 @@
                 }
                 free(path);
 
-                if (r >= 0 && UNIT(s)->load_state == UNIT_STUB) {
-                        /* Try Frugalware style rc.* init scripts */
-
-                        path = strjoin(*p, "/rc.", name, NULL);
-                        if (!path)
-                                return -ENOMEM;
-
-                        /* Drop .service suffix */
-                        path[strlen(path)-8] = 0;
-                        r = service_load_sysv_path(s, path);
-                        free(path);
-                }
-
                 if (r < 0)
                         return r;
 
